test/controller.test.ts

```diff
@@ -1981,6 +1966,536 @@ contract('Controller', ([owner, accountOwner1, accountOperator1, random]) => {
+              owner: accountOwner1,
+              sender: accountOwner1,
+              asset: expiredShortOtoken.address,
+              vaultId: vaultCounterBefore.toNumber() + 1,
+              amount: amountToMint.toNumber(),
+              index: '0',
+              data: ZERO_ADDR,
+            },
+            {
+              actionType: ActionType.DepositCollateral,
+              owner: accountOwner1,
+              sender: accountOwner1,
+              asset: usdc.address,
+              vaultId: vaultCounterBefore.toNumber() + 1,
+              amount: collateralToDeposit.toNumber(),
+              index: '0',
+              data: ZERO_ADDR,
+            },
+          ]
+
+          const marginPoolBalanceBefore = new BigNumber(await usdc.balanceOf(marginPool.address))
+          const senderBalanceBefore = new BigNumber(await usdc.balanceOf(accountOwner1))
+
+          await usdc.approve(marginPool.address, collateralToDeposit, {from: accountOwner1})
+          await controller.operate(actionArgs, {from: accountOwner1})
+
+          const marginPoolBalanceAfter = new BigNumber(await usdc.balanceOf(marginPool.address))
+          const senderBalanceAfter = new BigNumber(await usdc.balanceOf(accountOwner1))
+
+          assert.equal(
+            marginPoolBalanceAfter.minus(marginPoolBalanceBefore).toString(),
+            collateralToDeposit.toString(),
+            'Margin pool collateral asset balance mismatch',
+          )
+          assert.equal(
+            senderBalanceBefore.minus(senderBalanceAfter).toString(),
+            collateralToDeposit.toString(),
+            'Sender collateral asset balance mismatch',
+          )
+        })
+
+        it('should revert burning an expired long otoken', async () => {
+          // increment time after expiredLongOtoken expiry
+          await time.increase(3601) // increase time with one hour in seconds
+
+          const vaultId = new BigNumber(await controller.getAccountVaultCounter(accountOwner1))
+          const vault = await controller.getVault(accountOwner1, vaultId)
+          const shortAmountToBurn = new BigNumber('1')
+          const actionArgs = [
+            {
+              actionType: ActionType.BurnShortOption,
+              owner: accountOwner1,
+              sender: accountOwner1,
+              asset: expiredShortOtoken.address,
+              vaultId: vaultId.toNumber(),
+              amount: shortAmountToBurn.toNumber(),
+              index: '0',
+              data: ZERO_ADDR,
+            },
+          ]
+
+          assert.equal(await controller.isExpired(expiredShortOtoken.address), true, 'Long otoken is not expired yet')
+
+          await expectRevert(
+            controller.operate(actionArgs, {from: accountOwner1}),
+            'Controller: can not burn expired otoken',
+          )
+        })
+      })
+    })
+  })
+
+  describe('Exercise', () => {
+    let shortOtoken: MockOtokenInstance
+
+    before(async () => {
```

Comment:
separate into before and an `it` block. If you have assert.equal then you're testing something, you should put that in a `it` block